joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -938,10 +967,46 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
   resize_joint_trajectory_point(
     last_commanded_state_, dof_, std::numeric_limits<double>::quiet_NaN());
 
+  // create services
   query_state_srv_ = get_node()->create_service<control_msgs::srv::QueryTrajectoryState>(
     std::string(get_node()->get_name()) + "/query_state",
     std::bind(&JointTrajectoryController::query_state_service, this, _1, _2));
 
+  if (
+    !has_velocity_command_interface_ && !has_acceleration_command_interface_ &&
+    !has_effort_command_interface_)
+  {
+    auto qos = rclcpp::SystemDefaultsQoS();
+    qos.transient_local();
+    scaling_factor_sub_ = get_node()->create_subscription<SpeedScalingMsg>(
+      "~/speed_scaling_input", qos,
```

Comment:
```suggestion
      "~/speed_scaling", qos,
```
May be only speed_scaling is enough? As it is a subscriber, it is implicit. Any opinion?

Reply:
As the speed scaling can also be read from a state interface, it might make sense to be explicit here. But that's not really a strong opinion, I would also be fine with having `~/speed_scaling`

Reply:
@christophfroehlich any opinion?